Clarify that a contract without `receive` can still receive Ether through payable function calls (#13631)
ethereum/solidity: docs/contracts/functions.rst

function exists, but a payable :ref:`fallback function <fallback-function>`
exists, the fallback function will be called on a plain Ether transfer. If
neither a receive Ether nor a payable fallback function is present, the

> contract cannot receive Ether through transactions that do not match a payable function and throws an

Reviewer: ```suggestion
contract can still receive Ether through transactions that match a payable function. 
Otherwise, in the absence of all the above, the contract cannot receive Ether and
throws an exception.
```
Maybe this is more clear.

> contract cannot receive Ether through transactions that do not match a payable function and throws an

Reviewer 2: I'd adjust the wording a bit. It might not be clear what it means for the whole transaction to "match". We usually say that referring specifically to a selector.
```suggestion
contract cannot receive Ether through a transaction that does not represent a payable function call and throws an
```
